bundles/org.eclipse.equinox.event/src/org/eclipse/equinox/internal/event/EventComponent.java

```diff
@@ -10,25 +10,30 @@
  *
  * Contributors:
  *     IBM Corporation - initial API and implementation
+ *     Christoph Läubrich - switch to annotations
  *******************************************************************************/
 package org.eclipse.equinox.internal.event;
 
 import org.eclipse.equinox.internal.event.mapper.EventRedeliverer;
 import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.*;
 import org.osgi.service.event.Event;
 import org.osgi.service.event.EventAdmin;
 
+@Component(service = EventAdmin.class)
```

Comment:
It would also be a good idea to add the ProvideCapability for the EventAdmin.
```java
@Capability(namespace = ImplementationNamespace.IMPLEMENTATION_NAMESPACE, name = EventConstants.EVENT_ADMIN_IMPLEMENTATION, uses = Event.class, version = EventConstants.EVENT_ADMIN_SPECIFICATION_VERSION)
```

Reply:
> You don't need to declare a service here. If the Component directly implements an Interface, this will automatically become a service.

I know, but often found that when one evolves a component it is easily forgotten, so if one adds another interface the service is "suddenly" not there anymore and then people are confused, so I just configure it explicitly, you write code only once but read it often, so its not a big workload here :-)

> It would also be a good idea to add the ProvideCapability for the EventAdmin.

Good idea, I'll update that...